src/modules/top-level.nix

```
@@ -190,8 +190,11 @@ in
       fi
 
       mkdir -p .devenv
-      rm -f .devenv/profile
-      ln -s ${profile} .devenv/profile
+      if [ ! -L .devenv/profile ] || [ "$(${pkgs.coreutils}/bin/readlink .devenv/profile)" != "${profile}" ]
+      then
+        rm -f .devenv/profile
+        ln -s ${profile} .devenv/profile
```

Maybe instead? I was hoping it was atomic but looking at strace I see `symlinkat` followed by `reneameat` so not really. At least this way its in c and not bash?
```suggestion
        ln -nsf ${profile} .devenv/profile
```